golang/connection.go

```diff
@@ -175,6 +178,10 @@ func newConnection(conn net.Conn, initialState connectionState, handlers *handle
 		framePool = DefaultFramePool
 	}
 
+	connID := atomic.AddUint32(&nextConnID, 1)
+	log.SetPrefix(fmt.Sprintf("C%v ", connID))
```

Comment:
Log is shared between connections, no?  Seems like this needs to return a logger that has the prefix set, so that each connection gets a unique prefixed logger.